js/core/utils/data.js

```diff
@@ -131,14 +131,21 @@ var compileSetter = function(expr) {
     expr = bracketsToDots(expr);
 
     var pos = expr.lastIndexOf("."),
-        targetGetter = compileGetter(expr.substr(0, pos)),
+        targetGetterPath = expr.substr(0, pos),
+        targetGetter = compileGetter(targetGetterPath),
         targetPropName = expr.substr(1 + pos);
 
     return function(obj, value, options) {
         options = prepareOptions(options);
 
-        var target = targetGetter(obj, { functionsAsIs: options.functionsAsIs, unwrapObservables: options.unwrapObservables }),
-            prevTargetValue = readPropValue(target, targetPropName);
+        var target = targetGetter(obj, { functionsAsIs: options.functionsAsIs, unwrapObservables: options.unwrapObservables });
+
+        if(target === undefined) {
+            target = {};
+            compileSetter(targetGetterPath)(unwrapVariable(obj), target);
```

Comment:
unwrapVariable(obj) -> unwrap(obj, options)?

Reply:
I suppose it's very unlikely case, but I'll add the `options` to be on the safe side